docs/api/specs/desktopAgentCommunicationProtocol.md

```diff
@@ -0,0 +1,457 @@
+---
+
+# Desktop Agent Communication Protocol (DACP)
+
+:::info _[@experimental](../fdc3-compliance#experimental-features)_
+
+The Desktop Agent Communication Protocol (DACP) is an experimental feature added to FDC3 in 2.2. Limited aspects of its design may change in future versions and it is exempted from the FDC3 Standard's normal versioning and deprecation polices in order to facilitate any necessary change.
+
+:::
+
+The Desktop Agent Communication Protocol (DACP) constitutes a set of standardized JSON messages or 'wire protocol' that can be used to implement an interface to a Desktop Agent, encompassing all API calls events defined in the [Desktop Agent API](../ref/DesktopAgent.md). For example, the DACP is used by the [`@finos/fdc3` npm module](https://www.npmjs.com/package/@finos/fdc3) to communicate with Browser-Resident Desktop Agents or a connection setup via the [FDC3 Web Connection Protocol](./webConnectionProtocol).
+
+## Protocol conventions
+
+DACP messages are defined in [JSON Schema](https://json-schema.org/) in the [FDC3 github repository](https://github.com/finos/FDC3/tree/fdc3-for-web/schemas/api).
+
+:::tip
+
+TypeScript types representing all DACP and WCP messages are generated from the JSON Schema source and can be imported from the [`@finos/fdc3` npm module](https://www.npmjs.com/package/@finos/fdc3):
+
+```ts
+import { BrowserTypes } from '@finos/fdc3';
+```
+
+:::
+
+The protocol is composed of several different classes of message, each governed by a message schema:
+
+1. **App Request Messages** ([`AppRequest` schema](https://fdc3.finos.org/schemas/next/api/appRequest.schema.json)):
+    - Messages sent by an application representing an API call, such as [`DesktopAgent.broadcast`](../ref/DesktopAgent#broadcast), [`Channel.addContextListener`](../ref/Channel#addcontextlistener), or [`Listener.unsubscribe`](../ref/Types#listener).
+    - Message names all end in 'Request'.
+    - Each instance of a request message sent is uniquely identified by a `meta.requestUuid` field.
+
+2. **Agent Response Messages** ([`AgentResponse` schema](https://fdc3.finos.org/schemas/next/api/agentResponse.schema.json)):
+    - Response messages sent from the DA to the application, each relating to a corresponding _App Request Message_.
+    - Message names all end in 'Response'.
+    - Each instance of an Agent Response Message is uniquely identified by a `meta.responseUuid` field.
+    - Each instance of an Agent Response Message quotes the `meta.requestUuid` value of the message it is responding to.
+
+3. **Agent Event Messages** ([`AgentEvent` schema](https://fdc3.finos.org/schemas/next/api/agentEvent.schema.json)):
+    - Messages sent from the DA to the application that are due to actions in other applications, such as an inbound context resulting from another app's broadcast.
+    - Message names all end in 'Event'.
+    - Each instance of an Agent Response Message is uniquely identified by a `meta.eventUuid` field.
+
```

Comment:
For completeness, should there now be a 4. **Agent Acknowledgment Messages** specified?
i.e. Message names all end in 'Acknowledgment'
... which _can_ be used to acknowledge events where appropriate.

I wasn't sure whether this was left out because you originally had a heartbeatResponse or something (instead of heartbeatAcknowledgment) when this was originally written, and hence there was no 4th type at the time?  Or because there's currently only one example of this particular message type (heartbeatAcknowledgment)?

Reply:
HeartbeatAcknowledgeRequest is an appRequest message. I've updated the name/id to include Request for consistency, the type name already included it